apps/emqx/src/emqx_frame.erl

```diff
@@ -260,42 +271,52 @@ parse_packet(
-    ConnPacket = #mqtt_packet_connect{
-        proto_name = ProtoName,
-        proto_ver = ProtoVer,
-        is_bridge = (BridgeTag =:= 8),
-        clean_start = bool(CleanStart),
-        will_flag = bool(WillFlag),
-        will_qos = WillQoS,
-        will_retain = bool(WillRetain),
-        keepalive = KeepAlive,
-        properties = Properties,
-        clientid = ClientId
-    },
-    {ConnPacket1, Rest5} = parse_will_message(ConnPacket, Rest4, StrictMode),
-    {Username, Rest6} = parse_utf8_string(Rest5, StrictMode, bool(UsernameFlag)),
-    {Password, <<>>} = parse_utf8_string(Rest6, StrictMode, bool(PasswordFlag)),
-    ConnPacket1#mqtt_packet_connect{username = Username, password = Password};
+    case parse_utf8_string(FrameBin, StrictMode) of
+        % Note: Crash when reserved flag doesn't equal to 0, there is no strict
+        % compliance with the MQTT5.0.
+        {ProtoName,
+            <<BridgeTag:4, ProtoVer:4, UsernameFlag:1, PasswordFlag:1, WillRetain:1, WillQoS:2,
+                WillFlag:1, CleanStart:1, 0:1, KeepAlive:16/big, Rest2/binary>>} ->
+            {Properties, Rest3} = parse_properties(Rest2, ProtoVer, StrictMode),
+            {ClientId, Rest4} = parse_utf8_string(Rest3, StrictMode),
+            ConnPacket = #mqtt_packet_connect{
+                proto_name = ProtoName,
+                proto_ver = ProtoVer,
+                is_bridge = (BridgeTag =:= 8),
+                clean_start = bool(CleanStart),
+                will_flag = bool(WillFlag),
+                will_qos = WillQoS,
+                will_retain = bool(WillRetain),
+                keepalive = KeepAlive,
+                properties = Properties,
+                clientid = ClientId
+            },
+            {ConnPacket1, Rest5} = parse_will_message(ConnPacket, Rest4, StrictMode),
+            {Username, Rest6} = parse_utf8_string(Rest5, StrictMode, bool(UsernameFlag)),
+            case parse_utf8_string(Rest6, StrictMode, bool(PasswordFlag)) of
+                {Password, <<>>} ->
+                    ConnPacket1#mqtt_packet_connect{username = Username, password = Password};
+                _ ->
+                    ?PARSE_ERR(unexpected_data)
+            end;
+        _ ->
+            ?PARSE_ERR(incorrect_data)
+    end;
 parse_packet(
     #mqtt_packet_header{type = ?CONNACK},
     <<AckFlags:8, ReasonCode:8, Rest/binary>>,
     #{version := Ver, strict_mode := StrictMode}
 ) ->
-    {Properties, <<>>} = parse_properties(Rest, Ver, StrictMode),
-    #mqtt_packet_connack{
-        ack_flags = AckFlags,
-        reason_code = ReasonCode,
-        properties = Properties
-    };
+    case parse_properties(Rest, Ver, StrictMode) of
+        {Properties, <<>>} ->
+            #mqtt_packet_connack{
+                ack_flags = AckFlags,
+                reason_code = ReasonCode,
+                properties = Properties
+            };
+        _ ->
+            ?PARSE_ERR(unexpected_properties_data)
```

Comment:
this clause is lack of test coverage ?
at least no test case covering this error reason.
```suggestion
            ?PARSE_ERR(malformed_properties)
```